fwprovider/acme/resource_acme_account.go

```diff
@@ -0,0 +1,258 @@
+			},
+			"tos": schema.StringAttribute{
+				Optional: true,
+			},
+		},
+	}
+}
+
+// Configure accesses the provider-configured Proxmox API client on behalf of the resource.
+func (r *acmeAccountResource) Configure(
+	_ context.Context,
+	req resource.ConfigureRequest,
+	resp *resource.ConfigureResponse,
+) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(proxmox.Client)
+	if ok {
+		r.client = *client.Cluster().ACME().Account()
+	} else {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *proxmox.Client, got: %T",
+				req.ProviderData),
+		)
+	}
+}
+
+// Create creates a new ACME account on the Proxmox cluster.
+func (r *acmeAccountResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan acmeAccountModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createRequest := &account.ACMEAccountCreateRequestBody{}
+	createRequest.Contact = plan.Contact.ValueString()
+	createRequest.Directory = plan.Directory.ValueString()
+	createRequest.EABHMACKey = plan.EABHMACKey.ValueString()
+	createRequest.EABKID = plan.EABKID.ValueString()
+	createRequest.Name = plan.Name.ValueString()
+	createRequest.TOS = plan.TOS.ValueString()
+
+	err := r.client.Create(ctx, createRequest)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Unable to create ACME account '%s'", plan.Name),
+			err.Error(),
+		)
+
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read retrieves the current state of the ACME account from the Proxmox cluster.
+func (r *acmeAccountResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state acmeAccountModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	account, err := r.client.Get(ctx, state.Name.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Unable to read ACME account '%s'", state.Name),
+			err.Error(),
+		)
+
+		return
+	}
+
+	state.Directory = types.StringValue(account.Directory)
+	state.TOS = types.StringValue(account.TOS)
+	// XXX account.Location?
+	// XXX account.Account?
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update modifies an existing ACME account on the Proxmox cluster.
+func (r *acmeAccountResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan acmeAccountModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	updateRequest := &account.ACMEAccountUpdateRequestBody{}
+	updateRequest.Contact = plan.Contact.ValueString()
+	updateRequest.Name = plan.Name.ValueString() // XXX name can't be changed?
```

Comment:
`PUT` (https://pve.proxmox.com/pve-docs/api-viewer/#/cluster/acme/account/{name}) has two parameters, `contact` and `name`. But it looks like only `contact` can be changed, when trying to change `name` it says either that:
- the account was not found (makes sense, because `name` is in the path)
- or, if I use the old name in the path and the new name as body parameter, it says duplicate parameter.

I'm not sure how to deal with that in this resource.